desc/compute/_curve.py

```diff
@@ -366,20 +423,19 @@ def _x_FourierRZCurve(params, transforms, profiles, data, **kwargs):
     transforms={"R": [[0, 0, 0], [0, 0, 1]], "Z": [[0, 0, 1]], "grid": []},
     profiles=[],
     coordinates="s",
-    data=[],
+    data=["phi"],
     parameterization="desc.geometry.curve.FourierRZCurve",
 )
 def _x_s_FourierRZCurve(params, transforms, profiles, data, **kwargs):
     R0 = transforms["R"].transform(params["R_n"], dz=0)
     dR = transforms["R"].transform(params["R_n"], dz=1)
     dZ = transforms["Z"].transform(params["Z_n"], dz=1)
-    dphi = R0
-    coords = jnp.stack([dR, dphi, dZ], axis=1)
-    # convert to xyz for displacement and rotation
+    coords = jnp.stack([dR, R0, dZ], axis=1)
+    # convert to xyz for rotation using phi=s
     coords = rpz2xyz_vec(coords, phi=transforms["grid"].nodes[:, 2])
     coords = coords @ params["rotmat"].reshape((3, 3)).T
-    # convert back to rpz
-    coords = xyz2rpz_vec(coords, phi=transforms["grid"].nodes[:, 2])
+    # convert back to rpz using real phi to account for displacement
```

Comment:
We use the curve parameter `phi=s` to convert from rpz -> xyz because the derivatives `coords = [dR, R0, dZ]` assumes that the curve is in the standard orientation centered on the origin. When we convert back xyz -> rpz we need to account for possible rotation/translation from the `shift` and `rotmat` params. So here we use `phi = data["phi"]` which is the true toroidal angle phi.